songDisplayComponent/__tests__/setupTests.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Adapter from 'enzyme-adapter-react-16';
+import {configure} from 'enzyme';
+import {shallow} from 'enzyme';
+import App from '../client/App';
+
+configure({adapter: new Adapter()});
+
+describe('Initial Load Tests', () => {
+  // Create shallow of App
+  let wrap;
+  it('Renders without crashing', () => {
+    wrap = shallow(<App />);
+  });
+  it('Has a play button', () => {
+    expect(wrap.containsMatchingElement(<div className='play-button'></div>));
+  });
+  it('Has a container for the song name', () => {
+    expect(
+      wrap.containsMatchingElement(<div className='song-name-container'></div>)
+    );
+  });
+  it('Has a container for the artist name', () => {
+    expect(
+      wrap.containsMatchingElement(
+        <div className='artist-name-container'></div>
+      )
+    );
+  });
+  it('Has a container for the date posted', () => {
+    expect(
+      wrap.containsMatchingElement(
+        <div className='date-posted-container'></div>
+      )
+    );
+  });
+  it('Has a container for the song tag', () => {
+    expect(
+      wrap.containsMatchingElement(<div className='tags-container'></div>)
+    );
+  });
+  it('Has a container for song art', () => {
+    expect(wrap.containsMatchingElement(<div className='album-art'></div>));
+  });
+  it('Has a container for the song playback graph/slider', () => {
+    expect(wrap.containsMatchingElement(<div className='song-player'></div>));
+  });
+  it('Matches snapshot', () => {
+    expect(wrap).toMatchSnapshot();
+  });
+});
+
+describe('State Tests', () => {
+  // Create shallow of App
+  const wrap = shallow(<App />);
+  it('Can set state', () => {
+    wrap.setState({testPasses: true});
+    expect(wrap.state('testPasses')).toEqual(true);
+  });
+});
+
```

Comment:
Nice job getting your tests done. How did you manage to get around the async error you were getting earlier?

Reply:
Thanks man. I cheated -- since Maia said to just get some tests in, I removed the XHR requests from componentDidMount. In the future, I believe we can have an alternate function that requests local data for test purposes but we'd still have to integrate some form of promises to handle the async